components/style/selector_map.rs

```diff
@@ -110,17 +109,18 @@ impl SelectorMap<Rule> {
     ///
     /// Extract matching rules as per element's ID, classes, tag name, etc..
     /// Sort the Rules at the end to maintain cascading order.
-    pub fn get_all_matching_rules<E, V, F>(&self,
-                                           element: &E,
-                                           rule_hash_target: &E,
-                                           matching_rules_list: &mut V,
-                                           context: &mut MatchingContext,
-                                           quirks_mode: QuirksMode,
-                                           flags_setter: &mut F,
-                                           cascade_level: CascadeLevel)
+    pub fn get_all_matching_rules<E, V, F, C>(&self,
+                                              element: &E,
+                                              rule_hash_target: &E,
+                                              matching_rules_list: &mut V,
+                                              context: &mut MatchingContext,
+                                              classes_and_ids_case_sensitivity: C,
+                                              flags_setter: &mut F,
+                                              cascade_level: CascadeLevel)
         where E: TElement,
               V: VecLike<ApplicableDeclarationBlock>,
               F: FnMut(&E, ElementSelectorFlags),
+              C: CaseSensitivity,
```

Comment:
Yes, this to avoid the "tubofish" `::<>` syntax since passing a argument value looks more natural. But @bholley said on IRC that making the type parameter explicit can be useful to show that different code is called. The macro would be a bit more tricky, but it might be doable.